### ISSUE TITLE
Remove the special mangling rule for question marks

### ISSUE THREAD

**Issue by Kodiologist** [hylang/hy#2452]
This was one mangling detail that I disagreed with @gilch about years ago, but I belatedly think he was right. The rule that `foo?` mangles to `is_foo` (instead of `hyx_fooXquestion_markX`) should probably go. What's convinced me is wanting to use question marks on some names and then deciding not to because of mangling:

- In [Infinitesimal Quest 2 + ε](https://github.com/hylang/simalq), items have a Boolean attribute `destroy-on-pickup` that indicate whether the item is destroyed when it's picked up. I could call it `destroy-on-pickup?`, but that would mangle to the nonsensical `is_destroy_on_pickup`.
- Likewise, `attack?`, meaning "should this monster attack?", would be misleadingly mangled to `is_attack`.

Another concern is that Hy's core macro `is-not` can be written `not?`, which looks like a unary predicate for some sort of object called a "not", instead of a binary function that indicates whether `a` is not `b`.

One motivation for the question-mark rule was that it could provide a Lispy alias for a Python name: if a library has a function `is_foo`, you can call it as `foo`. But this purpose was always undermined by [Python core's inconsistency between `isfoo` and `is_foo`](https://github.com/python/cpython/issues/73274).

Finally, there's something to be said for making mangling a little simpler, since it's pretty complicated.

**Comment by scauligi** [comment]
Oh, I totally forgot this was even still a feature, somehow in my head I thought we removed back when we removed `*earmuffs*`.

I'm on board with removing `is_X` <=> `X?`, primarily because we can't use it consistently with Python's stdlib names, so it just makes things a bit more confusing.

---

Alternatively, we can add a new rule that mangles to `should_X` with a second question mark;
the examples above would mangle to `should_destroy_on_pickup` and `should_attack`, which make a lot more sense. On the Hy side they would of course be `destroy-on-pickup??` and `attack??`.

<sub>(this is not a real suggestion)</sub>
